android/src/main/java/com/oney/WebRTCModule/GetUserMediaImpl.java

```diff
@@ -70,8 +70,8 @@ class GetUserMediaImpl {
         }
 
         if (camera2supported) {
-            Log.d(TAG, "Creating video capturer using Camera2 API.");
-            cameraEnumerator = new Camera2Enumerator(reactContext);
+            Log.d(TAG, "Creating video capturer using Camera2 API with UVC support.");
+            cameraEnumerator = new UVCCamera2Enumerator(reactContext);
```

Comment:
I'd like if the user could explicitly tell the module to use certain implementation (Camera2Enumerator vs UVCCamera2Enumartor). I'm not sure how to do this correctly, since i only wanted to add implementations and not change the other source code. What do you think?

Reply:
Great question.

Generally speaking I think it should work out of the box, so it Just Works (TM).

A question I have is: is there a chance we get duplicate devices? As in, the builtin enumerator would detect the camera, then the UVC one would proceed to detect the same. Is that possible at all?

Reply:
Relating to your question about duplicate devices:

Yes, i think there is a slight chance that the built-in Camera2Enumerator could provide the external camera as well (This depends on the device and the manufacturer as far as i know).

I don't see this as an issue by the following reasoning:
If the external camera is enumerated twice (ie. by built-in and this ad-hoc), it should work whether the user chooses one or the other. Built-in enumerated uvc device will use the built-in capturer and ad-hoc will use the ad-hoc, since these appear as two distinct devices.

Of course, optimally the ad-hoc enumerated device should not be added to the enumerated devices if the built-in detects it, but currently i don't know how to robustly filter it out since none of my phones support external camera directly.